R-package/R/utils.R

```diff
@@ -208,17 +220,15 @@ generate.cv.folds <- function(nfold, nrows, stratified, label, params) {
     #  - For regression, leave y numeric and do stratification by quantiles.
     if (exists('objective', where = params) &&
```

Comment:
I know this code is not new to this PR, but I think `exists()` is dangerous here. If someone explicitly set `objective = NULL`  in `params`, this would evaluate to TRUE.

![image](https://user-images.githubusercontent.com/7608904/98901991-6a1fa180-247a-11eb-9236-a0b53770fbcb.png)

Then when you pass `NULL` into `convert.labels()`, it will result in a hard-to-understand error.

![image](https://user-images.githubusercontent.com/7608904/98902039-86234300-247a-11eb-95fc-550bb54a693e.png)

Would you be open to a PR that fixes these uses of `exists()`?

Reply:
Yes, let's fix it in a follow-up PR.

Reply:
ok sounds good! I'd be happy to contribute that, I think it'll be straightforward.